k8s/node.go

```diff
@@ -98,3 +100,106 @@ func RemoveTaintFromNodeByKey(k8sClient *kubernetes.Clientset, nodeName, taintKe
 	}
 	return nil
 }
+
+func SyncLabels(k8sClient *kubernetes.Clientset, nodeName string, toAddLabels, toDelLabels map[string]string) error {
+	updated := false
+	var err error
+	var node *v1.Node
+	for retries := 0; retries <= 5; retries++ {
+		node, err = GetNode(k8sClient, nodeName)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				logrus.Debugf("[hosts] Can't find node by name [%s]", nodeName)
+				return nil
+			}
+			return err
+		}
+		// Delete Labels
+		for key := range toDelLabels {
+			if _, ok := node.Labels[key]; ok {
+				delete(node.Labels, key)
+			}
+		}
+		// ADD Labels
+		for key, value := range toAddLabels {
+			node.Labels[key] = value
+		}
+		_, err = k8sClient.CoreV1().Nodes().Update(node)
+		if err != nil {
+			logrus.Debugf("Error syncing labels for node [%s]: %v", node.Name, err)
+			time.Sleep(time.Second * 5)
+			continue
+		}
+		updated = true
+		break
+	}
+	if !updated {
+		return fmt.Errorf("Timeout waiting for labels to be synced for node [%s]: %v", node.Name, err)
+	}
+	return nil
+}
+
+func SyncTaints(k8sClient *kubernetes.Clientset, nodeName string, toAddTaints, toDelTaints []string) error {
+	updated := false
+	var err error
+	var node *v1.Node
+	for retries := 0; retries <= 5; retries++ {
+		node, err = GetNode(k8sClient, nodeName)
```

Comment:
can we extract the code from inside the for loop, to a separate method? so it is easier to read